test/e2e/storage/testsuites/snapshottable.go

```diff
@@ -401,12 +406,79 @@ func CreateSnapshotResource(sDriver SnapshottableTestDriver, config *PerTestConf
 		// to pre-provision the snapshot as we normally would using their API.
 		// We instead dynamically take a snapshot and create another snapshot using
 		// the first snapshot's snapshot handle.
-		ginkgo.Skip("Preprovisioned test not implemented")
-		ginkgo.By("taking a snapshot with deletion policy retain")
-		ginkgo.By("recording the volume handle and status.snapshotHandle")
-		ginkgo.By("deleting the snapshot and snapshot content") // TODO: test what happens when I have two snapshot content that refer to the same content
+		ginkgo.By("taking a dynamic snapshot")
+		r.Vs = getSnapshot(pvcName, pvcNamespace, r.Vsclass.GetName())
+		r.Vs, err = dc.Resource(SnapshotGVR).Namespace(r.Vs.GetNamespace()).Create(context.TODO(), r.Vs, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+
+		err = WaitForSnapshotReady(dc, r.Vs.GetNamespace(), r.Vs.GetName(), framework.Poll, framework.SnapshotCreateTimeout)
+		framework.ExpectNoError(err)
+
+		r.Vs, err = dc.Resource(SnapshotGVR).Namespace(r.Vs.GetNamespace()).Get(context.TODO(), r.Vs.GetName(), metav1.GetOptions{})
+
+		snapshotStatus := r.Vs.Object["status"].(map[string]interface{})
+		snapshotContentName := snapshotStatus["boundVolumeSnapshotContentName"].(string)
+		framework.Logf("received snapshotStatus %v", snapshotStatus)
+		framework.Logf("snapshotContentName %s", snapshotContentName)
+		framework.ExpectNoError(err)
```

Comment:
These steps of creating a dynamic snapshot and validation of basic properties is duplicated in both dynamic and pre-provisioned case. Can we create a helper function and use in both the cases?

Reply:
What if we moved the common portion out of the switch and changed the switch statement to an if statement which only checks if it is a prepv test case?

Reply:
Yes that works too. Put a comment for the if part saying this is additional step for pre-provisioned snapshot case